src/main/java/mate/academy/bookstore/service/UserServiceImpl.java

```diff
@@ -1,29 +1,44 @@
 package mate.academy.bookstore.service;
 
+import java.util.Set;
 import lombok.RequiredArgsConstructor;
 import mate.academy.bookstore.dto.user.UserRegistrationRequestDto;
 import mate.academy.bookstore.dto.user.UserResponseDto;
 import mate.academy.bookstore.excepion.RegistrationException;
 import mate.academy.bookstore.mapper.UserMapper;
+import mate.academy.bookstore.model.Role;
 import mate.academy.bookstore.model.User;
+import mate.academy.bookstore.repository.roles.RoleRepository;
 import mate.academy.bookstore.repository.user.UserRepository;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 public class UserServiceImpl implements UserService {
     private final UserRepository userRepository;
     private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
+    private final RoleRepository roleRepository;
 
     @Override
     public UserResponseDto register(UserRegistrationRequestDto requestDto)
             throws RegistrationException {
-        if (userRepository.existsByEmail(requestDto.email())) {
-            throw new RegistrationException("User with this email: "
-                    + requestDto.email()
-                    + " already exists");
+        if (userRepository.findByEmail(requestDto.email()).isPresent()) {
```

Comment:
It's better create boolean existsByEmail(String email); in UserRepo and use it here